core/src/test/java/org/apache/calcite/sql/type/RelDataTypeSystemTest.java

```diff
@@ -143,6 +145,31 @@ static class Fixture extends SqlTypeFixture {
     final SqlTypeFactoryImpl customTypeFactory = new SqlTypeFactoryImpl(new CustomTypeSystem());
   }
 
+  @Test void testSupportNegativeScale() {
+    final SqlTypeFactoryImpl customTypeFactory =
+        new SqlTypeFactoryImpl(new DelegatingTypeSystem(new CustomTypeSystem()) {
+          @Override public boolean supportsNegativeScale() {
+            return true;
+          }
+        });
+    RelDataType dataType = customTypeFactory.createSqlType(SqlTypeName.DECIMAL, 10, -5);
+    assertEquals(SqlTypeName.DECIMAL, dataType.getSqlTypeName());
+    assertEquals(10, dataType.getPrecision());
+    assertEquals(-5, dataType.getScale());
+  }
+
+  @Test void testNotSupportNegativeScale() {
```

Comment:
if you support a negative scale, is there a lower limit?
For example, in Oracle the scale has maximum and minimum values: https://docs.oracle.com/cd/B19306_01/server.102/b14200.pdf, page 44 in the pdf